From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Randy Eckenrode <randy@largeandhighquality.com>
Date: Thu, 24 Sep 2026 15:51:35 -0400
Subject: [PATCH 3/3] Add missing imports

This is apparently needed after updating to Glibc 2.44. In 6.4, the
code that uses `stat` moved from `URL_Swift.swift` to `URL_Impl.swift`.
---
 Sources/FoundationEssentials/Date.swift                         | 2 ++
 .../FoundationEssentials/FileManager/FileManager+Basics.swift   | 2 ++
 Sources/FoundationEssentials/URL/URL.swift                      | 2 ++
 Sources/FoundationEssentials/URL/URL_Impl.swift                 | 2 ++
 4 files changed, 8 insertions(+)

diff --git a/Sources/FoundationEssentials/Date.swift b/Sources/FoundationEssentials/Date.swift
index 45b4844..82b8711 100644
--- a/Sources/FoundationEssentials/Date.swift
+++ b/Sources/FoundationEssentials/Date.swift
@@ -24,6 +24,8 @@ import WinSDK
 @preconcurrency import WASILibc
 #endif
 
+internal import _FoundationCShims
+
 #if !FOUNDATION_FRAMEWORK
 /// A number of seconds.
 @available(macOS 10.10, iOS 8.0, watchOS 2.0, tvOS 9.0, *)
diff --git a/Sources/FoundationEssentials/FileManager/FileManager+Basics.swift b/Sources/FoundationEssentials/FileManager/FileManager+Basics.swift
index 96ea0d2..c66f939 100644
--- a/Sources/FoundationEssentials/FileManager/FileManager+Basics.swift
+++ b/Sources/FoundationEssentials/FileManager/FileManager+Basics.swift
@@ -25,6 +25,8 @@ import WinSDK
 @preconcurrency import WASILibc
 #endif
 
+internal import _FoundationCShims
+
 #if FOUNDATION_FRAMEWORK
 internal import Foundation_Private
 #endif
diff --git a/Sources/FoundationEssentials/URL/URL.swift b/Sources/FoundationEssentials/URL/URL.swift
index cfc5352..de00dab 100644
--- a/Sources/FoundationEssentials/URL/URL.swift
+++ b/Sources/FoundationEssentials/URL/URL.swift
@@ -13,6 +13,8 @@
 public struct URLResourceKey {}
 #endif
 
+internal import _FoundationCShims
+
 #if FOUNDATION_FRAMEWORK
 internal import _ForSwiftFoundation
 internal import CoreFoundation_Private.CFURL
diff --git a/Sources/FoundationEssentials/URL/URL_Impl.swift b/Sources/FoundationEssentials/URL/URL_Impl.swift
index 91ed72c..6e5321b 100644
--- a/Sources/FoundationEssentials/URL/URL_Impl.swift
+++ b/Sources/FoundationEssentials/URL/URL_Impl.swift
@@ -24,6 +24,8 @@ import WinSDK
 @preconcurrency import WASILibc
 #endif
 
+internal import _FoundationCShims
+
 #if FOUNDATION_FRAMEWORK
 internal import _ForSwiftFoundation
 internal import Synchronization
